src/kindtech/geo/_arcgis/_ingestion.py

```diff
@@ -0,0 +1,185 @@
+"""
+ArcGIS REST Services ingestion module.
+
+This module provides functions to load ArcGIS REST Services data from the web and save
+it to a CSV file
+
+The CSV file is later used as the main lookup table to find the relevant data
+"""
+
+import calendar
+import logging
+from pathlib import Path
+from typing import Any
+
+import polars as pl
+import requests
+from bs4 import BeautifulSoup
+
+# Set up logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+# Constants
+ARCGIS_BASE_URL = "https://services1.arcgis.com/ESMARspQHYMw9BZ9/arcgis/rest/services/"
+DEFAULT_OUTPUT_PATH = Path(__file__).parent / "data" / "arcgis_services.csv"
+
+# Lookup tables for parsing
+MONTH = {}
+for idx, name in enumerate(calendar.month_name):  # '' then 'January'…
+    if name:
+        MONTH[name.upper()[:3]] = idx  # JAN, FEB, …
+        MONTH[name.upper()] = idx  # JANUARY, FEBRUARY …
+
+REGION = {"UK", "GB", "EW"}
+RES = {"BFC", "BFE", "BGC", "BSC"}
+DROP = {
+    "LOCAL",
+    "AUTHORITY",
+    "DISTRICT",
+    "DISTRICTS",
+    "BOUNDARY",
+    "BOUNDARIES",
+    "WITH",
+    "AND",
+    "IN",
+}
+
+
+def _load_arcgis_data(url: str = ARCGIS_BASE_URL) -> str:
+    """
+    Load ArcGIS REST Services catalog HTML from the given URL.
+
+    Args:
+        url: The URL of the ArcGIS REST Services catalog.
+
+    Returns:
+        The HTML content of the catalog page.
+
+    Raises:
+        requests.RequestException: If the request fails.
+    """
+    logger.info(f"Loading ArcGIS data from {url}")
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise exception for 4XX/5XX responses
+        return response.text
+    except requests.RequestException as e:
+        logger.error(f"Failed to load ArcGIS data: {e}")
+        raise
+
+
+def _parse_html_for_services(html_content: str) -> list[dict[str, Any]]:
+    """
+    Parse the HTML content to extract FeatureServer services and their metadata
+
+    Args:
+        html_content: The HTML content to parse.
+
+    Returns:
+        List of dictionaries containing service metadata (year, month, region,
+        resolution, raw name)
+    """
+    logger.info("Parsing HTML for services")
+    kept = []
+
+    soup = BeautifulSoup(html_content, "html.parser")
+    services_list = soup.find("ul", id="servicesList")
+
+    if not services_list:
+        logger.warning("No services list found in HTML content")
+        return kept
+
+    for li in services_list.find_all("li"):
+        link = li.find("a")
+        if link and "FeatureServer" in link["href"]:
+            service_name = link.text
+
+            kept.append(service_name)
+
+    logger.info(f"Total FeatureServer services found: {len(kept)}")
+
+    return kept
+
+
+def _process_and_save(services: list):
+    df = pl.DataFrame({"arcgis_id": services})
+
+    print(df.height)
+
+    to_skip = [
```

Comment:
quite a lot of custom logic here - were you thinking that we can use a CI job to check for these periodically?

Reply:
On custom logic, yes I don't really have a better solution for it, it was hard(er) to write custom refer that could handle these edge cases: 

- RUC (rural 
- 1961 census map developed in 2022 as a one off